src/runtime/auto-prompt-manager-test.js

```diff
@@ -907,4 +974,228 @@ describes.realWin('AutoPromptManager', {}, (env) => {
+        storedImpressions,
+        storedDismissals,
+        'contribution,TYPE_REGISTRATION_WALL',
+        2
+      );
+      miniPromptApiMock.expects('create').never();
+
+      await autoPromptManager.showAutoPrompt({
+        autoPromptType: AutoPromptType.CONTRIBUTION,
+        alwaysShow: false,
+        displayLargePromptFn: alternatePromptSpy,
+      });
+      await tick(2);
+
+      expect(startSpy).to.have.been.calledOnce;
+      expect(actionFlowSpy).to.have.been.calledWith(deps, {
+        action: 'TYPE_NEWSLETTER_SIGNUP',
+        onCancel: sandbox.match.any,
+        autoPromptType: AutoPromptType.CONTRIBUTION,
+      });
+      expect(alternatePromptSpy).to.not.have.been.called;
+      expect(autoPromptManager.promptDisplayed_).to.equal(
+        'TYPE_NEWSLETTER_SIGNUP'
+      );
+      await verifyOnCancelStores(
+        'contribution,TYPE_REGISTRATION_WALL,TYPE_NEWSLETTER_SIGNUP'
+      );
+    });
+
+    it('should show nothing if the the last Audience Action was previously dismissed and is not in the next Contribution prompt time', async () => {
+      const storedImpressions = (CURRENT_TIME - 5).toString();
+      const storedDismissals = (CURRENT_TIME - 10).toString();
+      setupPreviousImpressionAndDismissals(
+        storedImpressions,
+        storedDismissals,
+        'contribution,TYPE_REGISTRATION_WALL,TYPE_NEWSLETTER_SIGNUP',
+        1
+      );
+      miniPromptApiMock.expects('create').never();
+
+      await autoPromptManager.showAutoPrompt({
+        autoPromptType: AutoPromptType.CONTRIBUTION,
+        alwaysShow: false,
+        displayLargePromptFn: alternatePromptSpy,
+      });
+      await tick(2);
+
+      expect(startSpy).to.not.have.been.called;
+      expect(actionFlowSpy).to.not.have.been.called;
+      expect(alternatePromptSpy).to.not.have.been.called;
+      expect(autoPromptManager.promptDisplayed_).to.equal(null);
+    });
+
+    it('should show the Contribution Flow even if there is an available Audience Action that was previously dismissed and is in the next Contribution prompt time', async () => {
+      // One stored impression from 20s ago and one dismissal from 6s ago.
+      const storedImpressions = (CURRENT_TIME - 20000).toString();
+      const storedDismissals = (CURRENT_TIME - 6000).toString();
+      setupPreviousImpressionAndDismissals(
+        storedImpressions,
+        storedDismissals,
+        'contribution,TYPE_REGISTRATION_WALL',
+        1
+      );
+      miniPromptApiMock.expects('create').once();
+
+      await autoPromptManager.showAutoPrompt({
+        autoPromptType: AutoPromptType.CONTRIBUTION,
+        alwaysShow: false,
+        displayLargePromptFn: alternatePromptSpy,
+      });
+      await tick(2);
+
+      expect(startSpy).to.not.have.been.called;
+      expect(actionFlowSpy).to.not.have.been.called;
+      expect(alternatePromptSpy).to.not.have.been.called;
+      expect(autoPromptManager.promptDisplayed_).to.equal(null);
```

Comment:
> expect(autoPromptManager.promptDisplayed_).to.equal(null);
Should this be equal(AutoPromptType.CONTRIBUTION)?

Reply:
That would mean we would keep appending `contribution` to the dismissals, which I wouldn't want. Only need to do it the one time

Reply:
Hmm, not sure I follow.

I might miss something: It seems `this.promptDisplayed_` and `localStorage[dismissedprompts]` never get reset to null when it is in the next Contribution prompt time. How do you prevent `contribution` is not being appended to the dismissals?

Also, for this setup, if the user dismisses the contribution prompt, will we show regwall or newsletter next since the dismissals right now is `contribution,TYPE_REGISTRATION_WALL`? We might probably want to add a test case for it?

Reply:
> How do you prevent contribution from being appended to the dismissals?

Oh nvm, I found that `this.promptDisplayed_` is only set to `contribution` if the dismissals is null.

Reply:
>Also, for this setup, if the user dismisses the contribution prompt, will we show regwall or newsletter next since the dismissals right now is contribution,TYPE_REGISTRATION_WALL? We might probably want to add a test case for it?

Synced offline. In this case, newsletter should be shown next. This followed the product requirement: "if they dismissed a newsletter, they probably just never want to give any other info to the publisher".